src/main/java/com/stripe/net/LiveStripeResponseGetter.java

```diff
@@ -150,6 +138,34 @@ static String formatAppInfo(Map<String, String> info) {
     return headers;
   }
 
+  private static Map<String, String> buildClientUserAgentMap() {
+    String[] propertyNames = {"os.name", "os.version", "os.arch", "java.version", "java.vendor",
+      "java.vm.version", "java.vm.vendor"};
+    Map<String, String> clientUserAgentMap = new HashMap<>();
+    for (String propertyName : propertyNames) {
+      clientUserAgentMap.put(propertyName, System.getProperty(propertyName));
+    }
+
+    clientUserAgentMap.put("bindings.version", Stripe.VERSION);
+    clientUserAgentMap.put("lang", "Java");
+    clientUserAgentMap.put("publisher", "Stripe");
+    if (Stripe.getAppInfo() != null) {
+      clientUserAgentMap.put("application", ApiResource.GSON.toJson(Stripe.getAppInfo()));
+    }
+
+    // Format keys to be friendly for log analysis and consistent with other client libraries
+    // Key containing '.' is considered nested in Splunk, but that's not the case here,
+    // so we are replacing it with '_'
+    Set<String> keySet = new HashSet<>(clientUserAgentMap.keySet());
+    for (String key: keySet) {
+      String value = clientUserAgentMap.get(key);
+      clientUserAgentMap.remove(key);
+      String formattedKey = key.replace('.', '_');
```

Comment:
This library's programming style dictates that this variable definition should be at the top of this block, after the line that declares `value`. Can you either move this line up omit the variable and put the `key.replace()` call directly onto the `clientUserAgentMap.put()` call?

Reply:
Moved the line up to the top of this block.
Thanks Fred for the careful review!